magda-storage-api/src/test/storage.spec.ts

```diff
@@ -0,0 +1,95 @@
+import {} from "mocha";
+import sinon from "sinon";
+import express from "express";
+import _ from "lodash";
+import request from "supertest";
+
+import createApiRouter from "../createApiRouter";
+import MagdaMinioClient from "../MagdaMinioClient";
+
+const Minio = require("minio");
+
+describe("Storage API tests", () => {
+    let app: express.Application;
+    const minioClientOpts = {
+        endPoint: process.env["MINIO_HOST"],
+        port: Number(process.env["MINIO_PORT"]),
+        useSSL: false,
+        accessKey: process.env["MINIO_ACCESS_KEY"],
+        secretKey: process.env["MINIO_SECRET_KEY"],
+        bucket: "magda-test-1"
+    };
+    const minioClient = new Minio.Client(minioClientOpts);
+
+    before(() => {
+        minioClient.makeBucket(minioClientOpts.bucket, function(err: Error) {
+            if (err) {
+                return console.log("Error creating bucket.", err);
+            }
+            console.log('Bucket created successfully in "us-east-1".');
+        });
+    });
+
+    beforeEach(() => {
+        app = express();
+        app.use(
+            "/v0",
+            createApiRouter({
+                objectStoreClient: new MagdaMinioClient(minioClientOpts)
+            })
+        );
+    });
+
+    afterEach(() => {
+        if ((<sinon.SinonStub>console.error).restore) {
+            (<sinon.SinonStub>console.error).restore();
+        }
+    });
+
+    describe("Upload", () => {
+        it("Uploading a simple file", () => {
+            return request(app)
+                .put("/v0/upload-test-file")
+                .set("Accept", "application/json")
+                .set("Content-Type", "text/plain")
+                .send("LALALALALALALALALA")
+                .expect(200);
+        });
+    });
+
+    describe("Download", () => {
+        it("Uploading and then downloading the simple file", function() {
+            return request(app)
+                .put("/v0/download-test-file-1")
+                .set("Accept", "application/json")
+                .set("Content-Type", "text/csv")
+                .send("Testing download")
+                .expect(200)
+                .then(_res => {
+                    return request(app)
+                        .get("/v0/download-test-file-1")
+                        .set("Accept", "application/json")
+                        .set("Accept", "text/plain")
+                        .expect(200)
```

Comment:
This is good but it'd be good to test that downloading tests the correct content type. Probably having one test for binary and one test for text would be good too :).

Reply:
Added a test for binary files :+1: